torchtune/models/llama3/_tokenizer.py

```diff
@@ -166,11 +182,18 @@ def decode(
         Returns:
             str: The decoded string.
         """
-        return self.tt_model.decode(
-            token_ids,
+        # We will remove special tokens manually via regex on the decoded string.
+        # This is because removing all special tokens does not remove the role and
+        # whitespace added from the special tokens, i.e., the "user" and "\n\n" in
+        # "<|start_header_id|>user<|end_header_id|>\n\n"
```

Comment:
I would maybe move this comment up to where you define `self._special_token_regex` and `self._special_token_header_regex`

Reply:
this is where it actually happens, so makes more sense to keep it here? no strong opinions

Reply:
Yeah I feel the same, fine to keep it here then